core/src/main/java/org/pentaho/di/core/row/ValueDataUtil.java

```diff
@@ -415,30 +491,62 @@ public static Long ChecksumCRC32( ValueMetaInterface metaA, Object dataA ) {
 
       checksum = cis.getChecksum().getValue();
 
+    } catch ( KettleFileNotFoundException e ) {
+      if ( failIfNoFile ) {
+        throw e;
+      }
     } catch ( Exception e ) {
-      // ignore - should likely log the exception
+      log.debug( e.getMessage() );
     } finally {
-      if ( file != null ) {
-        try {
-          file.close();
-          file = null;
-        } catch ( Exception e ) {
-          // Ignore
-        }
-      }
+      IOUtils.closeQuietly( file );
     }
     return checksum;
   }
 
+  /**
+   * @deprecated Use {@link ValueDataUtil#checksumAdler32(ValueMetaInterface, Object, boolean)} instead
+   */
+  @Deprecated
   public static Long ChecksumAdler32( ValueMetaInterface metaA, Object dataA ) {
     long checksum = 0;
+    try {
+      checksum = checksumAdler32( metaA, dataA, false );
+    } catch ( KettleFileNotFoundException e ) {
+      // Ignore
+    }
+    return checksum;
+  }
+
+  /**
+   *
+   * @param metaA
+   *   The ValueMetaInterface
+   * @param dataA
+   *   Filename
+   * @param failIfNoFile
+   *   Indicates if the transformation should fail if no file is found
+   * @return File's Adler32 checksum
+   * @throws KettleFileNotFoundException
+   */
+  public static Long checksumAdler32( ValueMetaInterface metaA, Object dataA, boolean failIfNoFile )
+          throws KettleFileNotFoundException {
+    long checksum = 0;
+
+    if ( dataA == null ) {
```

Comment:
This actually is not a file not found.... it's a file path/name not provided.... I'm not sure the file not found exception suits this case